tests/test_java.py

```diff
@@ -124,7 +69,8 @@ def test_java_async_profiler_cpu_mode(
         java_safemode=False,
         java_mode="ap",
     ) as profiler:
-        process_collapsed = profiler.snapshot().get(application_pid)
+        result = profiler.snapshot()
+        process_collapsed = result[next(iter(result.keys()))]
```

Comment:
Why? It made sense to check by PID

Reply:
Because the pid returned by `application_pid` for container doesn't match the pid of the java process within the container.

Reply:
Ah, due to the `sh -c` thing. Why did we do that btw?

Reply:
`sh -c exec java`?

Reply:
because we need the container to stay up after the java process inside is killed

Reply:
Gotcha

Reply:
Can we instead verify that there is exactly one result? It will be very confusing if we find multiple results and this leads the test to fail